### ISSUE TITLE
lsblk shows incorrect serial number

### ISSUE THREAD

**Issue by painhardcore** [karelzak/util-linux#1143]
Found with lsblk from util-linux 2.34
```
# lsblk -o NAME,SERIAL | grep sdc
sdc          5000cca0541ae674
```
Shows WWN instead of SN.
```
root:/# udevadm info --query=all --name=/dev/sdc | grep SERIAL
E: SCSI_IDENT_SERIAL=KJGATM4R
E: ID_SERIAL=35000cca0541ae674
E: ID_SERIAL_SHORT=5000cca0541ae674
```

Since ID_SCSI_SERIAL is not present it uses ID_SERIAL. 
It can be fixed by adding new optional value to find, so the fallback scheme will be ID_SCSI_SERIAL -> SCSI_IDENT_SERIAL -> ID_SERIAL

**Comment by karelzak** [comment]
The current order in lsblk is:
```
               else if (lookup(buf, "ID_SCSI_SERIAL", &prop->serial)) ;
               else if (lookup(buf, "ID_SERIAL_SHORT", &prop->serial)) ;
               else if (lookup(buf, "ID_SERIAL", &prop->serial)) ;
```

Where SCSI_IDENT_SERIAL does come from?  I don't see it in udev upstream rules.

No problem to add SCSI_IDENT_SERIAL, but I'd like to know more details, I see it the first time :-)

**Comment by zeha** [comment]
Appears to be an sg3_utils thing...

https://github.com/hreinecke/sg3_utils/blame/d82f040c69689305ca1d318d3dc0e1e42ab6ffa3/src/sg_inq.c#L3402

However sg3_utils also avoids overriding ID_SCSI_SERIAL (I guess to avoid changing by-id links?): https://github.com/hreinecke/sg3_utils/blob/2abec32d08e66f933c3ac2415c67460c6b210ba6/scripts/55-scsi-sg3_id.rules#L114

Related: https://bugzilla.redhat.com/show_bug.cgi?id=1328771

**Comment by karelzak** [comment]
Thanks. I have added SCSI_IDENT_SERIAL as the first possibility.

**Comment by karelzak** [comment]
Please, retest it with your environment and reopen it if necessary. Thanks for your report!

**Comment by painhardcore** [comment]
@karelzak thanks, but correct me if I'm wrong - you've added ID_SCSI_IDENT_SERIAL, not SCSI_IDENT_SERIAL.

**Comment by karelzak** [comment]
Ok, it's really without the prefix. Fixed.

**Comment by brian-lamb-software-engineer** [comment]
FYI on RHEL7 as of today im still experiencing this issue.  
```$ lsblk -d -o name,serial
NAME SERIAL
sda  50000c0f01e5dabc
sdb  5000c50093a8dfe7
sdc  5000c50093a9182b
sdd  5000c50084818db7
sde  5000c50084889cf3
sdf  5000c5008491a803
sdg  5000c50093a9052f
sdh  50000c0f01e63ff0
sdi  WD-WMASU0187755
```

**Comment by karelzak** [comment]
@brian-lamb-software-engineer "lsblk --version"?

**Comment by katotheo** [comment]
> 
```
xxx@yyy:~$ lsblk -o NAME,SERIAL --nodeps | grep sdb
sdb   5000c500dbd6e027
xxx@yyy:~$ udevadm info --query=all --name=/dev/sdb | grep SERIAL
E: SCSI_IDENT_SERIAL=ZL2KAKW2
E: ID_SERIAL=35000c500dbd6e027
E: ID_SERIAL_SHORT=5000c500dbd6e027
xxx@yyy:~$ lsblk --version
lsblk from util-linux 2.34
xxx@yyy:~$
```

**Comment by katotheo** [comment]
> @brian-lamb-software-engineer "lsblk --version"?

```
xxx@yyy:~$ lsblk -o NAME,SERIAL --nodeps | grep sdb
sdb   5000c500dbd6e027
xxx@yyy:~$ udevadm info --query=all --name=/dev/sdb | grep SERIAL
E: SCSI_IDENT_SERIAL=ZL2KAKW2
E: ID_SERIAL=35000c500dbd6e027
E: ID_SERIAL_SHORT=5000c500dbd6e027
xxx@yyy:~$ lsblk --version
lsblk from util-linux 2.34
xxx@yyy:~$
```

**Comment by karelzak** [comment]
Upgrade, fixed in version >= 2.37.

**Comment by katotheo** [comment]
> Upgrade, fixed in version >= 2.37.

Thanks
